src/ngraph/frontend/onnx_import/op/matmul_integer.cpp

```diff
@@ -0,0 +1,62 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "op/matmul_integer.hpp"
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/builder/quantization/quantized_linear_matmul.hpp"
+#include "ngraph/frontend/onnx_import/exceptions.hpp"
+
+using namespace ngraph::builder;
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector matmul_integer(const Node& node)
```

Comment:
This won't work for input tensors with rank > 2, since nGraph `Dot` doesn't work as NumPy's matmul. There should be a logic similar to this https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/frontend/onnx_import/op/matmul.cpp#L124 to handle higher order tensors.

Reply:
Added `MatmulFactory` classes to address this issue.